app/scripts/lib/validate.js

```diff
@@ -28,28 +44,27 @@ define(function (require, exports, module) {
      * @return true if email is valid, false otw.
      */
     isEmailValid: function (email) {
-      if (typeof email !== 'string') {
+      if (typeof email !== 'string' || email.length > 256) {
         return false;
       }
 
       var parts = email.split('@');
+      var localPart = parts[0];
+      var domainPart = parts[1];
+
+      if (parts.length !== 2 || localPart.length > 64) {
+        return false;
+      }
+
+      if (unicodeRegex.test(localPart)) {
+        localPart = punycode.toASCII(localPart);
```

Comment:
I believe all auth-server endpoints will accept utf8-encoded emails, validate by converting to punycode, but work with the original form of the email for all further operations like storing, sending, etc.